substrate-examples/offchain-worker/src/lib.rs

```diff
@@ -0,0 +1,780 @@
+            let current_block = <system::Pallet<T>>::block_number();
+            <NextUnsignedAt<T>>::put(current_block + T::UnsignedInterval::get());
+            Ok(().into())
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight({0})]
+        pub fn submit_price_unsigned_with_signed_payload(
+            origin: OriginFor<T>,
+            price_payload: PricePayload<T::Public, T::BlockNumber>,
+            _signature: T::Signature,
+        ) -> DispatchResultWithPostInfo {
+            // This ensures that the function can only be called via unsigned transaction.
+            ensure_none(origin)?;
+            // Add the price to the on-chain list, but mark it as coming from an empty address.
+            Self::add_price(None, price_payload.price);
+            // now increment the block number at which we expect next unsigned transaction.
+            let current_block = <system::Pallet<T>>::block_number();
+            <NextUnsignedAt<T>>::put(current_block + T::UnsignedInterval::get());
+            Ok(().into())
+        }
+
+        #[pallet::call_index(3)]
+		#[pallet::weight(100)]
+		pub fn include_data(origin: OriginFor<T>, _data: Vec<u8>) -> DispatchResult {
+			frame_system::ensure_signed(origin)?;
+			Ok(())
+		}
+    }
+
+    /// Events for the pallet.
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Event generated when new price is accepted to contribute to the average.
+        NewPrice { price: u32, maybe_who: Option<T::AccountId> },
+    }
+
+    #[pallet::validate_unsigned]
+    impl<T: Config> ValidateUnsigned for Pallet<T> {
+        type Call = Call<T>;
+
+        /// Validate unsigned call to this module.
+        ///
+        /// By default unsigned transactions are disallowed, but implementing the validator
+        /// here we make sure that some particular calls (the ones produced by offchain worker)
+        /// are being whitelisted and marked as valid.
+        fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {          
+            // Firstly let's check that we call the right function.
+            if let Call::submit_price_unsigned_with_signed_payload {
+                price_payload: ref payload,
+                ref signature,
+            } = call
+            {                
+                let signature_valid =
+                    SignedPayload::<T>::verify::<T::AuthorityId>(payload, signature.clone());
+                if !signature_valid {
+                    return InvalidTransaction::BadProof.into()
+                }
+                let res = Self::validate_transaction_parameters(&payload.block_number, &payload.price);
+                // Properly check that the transaction is valid before applying the data
+                if res.is_ok() {
+                    Self::check_data(&payload.block_number, &payload.price);
+                }

```

Comment:
As far as I understand this part is not checked by mirai, as the test call uses the submit_price_unsigned call